junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -146,9 +149,10 @@ private void assertSupportedType(String target, Class<?> type) {
 		}
 	}
 
-	private Object getPathOrFile(Class<?> type, ExtensionContext extensionContext) {
+	private Object getPathOrFile(Class<?> type, ExtensionContext extensionContext, String id) {
+		String key = id.isEmpty() ? KEY : KEY + '+' + id;
 		Path path = extensionContext.getStore(NAMESPACE) //
-				.getOrComputeIfAbsent(KEY, key -> createTempDir(), CloseablePath.class) //
+				.getOrComputeIfAbsent(key, __ -> createTempDir(), CloseablePath.class) //
```

Comment:
⭕ I liked your idea of using the key as a prefix for the temp dir (Maybe as a suffix to `TEMP_DIR_PREFIX`). We'd probably have to sanitize the string a bit for that, e.g. remove all non-alphanumeric characters. WDYT?

Reply:
Sanitizing is brittle at best: https://stackoverflow.com/a/31564306/1431016 -- not using the key as a prefix is the safe bet.

Reply:
**Team decision:** Let's not use the key as a prefix for now since sanitizing might be a rabbit hole.